skiko/src/jvmMain/objectiveC/macos/Drawlayer.mm

```diff
@@ -399,6 +399,18 @@ JNIEXPORT jint JNICALL Java_org_jetbrains_skiko_SystemTheme_1awtKt_getCurrentSys
     }
 }
 
+JNIEXPORT jlong JNICALL Java_org_jetbrains_skiko_HardwareLayer_getCurrentDPI(JNIEnv *env, jobject component, jlong platformInfoPtr)
+{
+    @autoreleasepool {
+        LayerHandler *layer = findByObject(env, component);
+        NSScreen *screen = [layer.window screen];
+        NSDictionary *description = [screen deviceDescription];
+        NSSize displayPixelSize = [[description objectForKey:NSDeviceSize] sizeValue];
+        CGSize displayPhysicalSize = CGDisplayScreenSize([[description objectForKey:@"NSScreenNumber"] unsignedIntValue]);
+        return (jlong)(displayPixelSize.width / (displayPhysicalSize.width / 25.4));
```

Comment:
Could you please explain `25.4`.

Reply:
dpi - dots per inch
the physical size of the display is in millimeters so we need to convert it to inches so we divide it by 25.4.

Reply:
For me on 27 inch iMac (height 20.3 inches  width of 25.6 inches ) at 5120 x 2880 this function gives 108 dot per inch, which almost matches claimed 110 dpi.